controller/ca/controller.go

```diff
@@ -0,0 +1,216 @@
+package ca
+
+import (
+	"time"
+
+	"github.com/runconduit/conduit/controller/k8s"
+	pkgK8s "github.com/runconduit/conduit/pkg/k8s"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type CertificateController struct {
+	namespace   string
+	k8sAPI      *k8s.API
+	syncHandler func(key string) error
+	queue       workqueue.RateLimitingInterface
+}
+
+func NewCertificateController(conduitNamespace string, k8sAPI *k8s.API) *CertificateController {
+	c := &CertificateController{
+		namespace: conduitNamespace,
+		k8sAPI:    k8sAPI,
+		queue: workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(), "certificates"),
+	}
+
+	k8sAPI.Pod().Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.handlePodAdd,
+			UpdateFunc: c.handlePodUpdate,
+		},
+	)
+
+	k8sAPI.CM().Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.handleConfigMapAdd,
+			UpdateFunc: c.handleConfigMapUpdate,
+			DeleteFunc: c.handleConfigMapDelete,
+		},
+	)
+
+	c.syncHandler = c.syncNamespace
+
+	return c
+}
+
+func (c *CertificateController) Run(stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	log.Info("starting certificate controller")
+	defer log.Info("shutting down certificate controller")
+
+	go wait.Until(c.worker, time.Second, stopCh)
+
+	<-stopCh
+}
+
+func (c *CertificateController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *CertificateController) processNextWorkItem() bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.syncHandler(key.(string))
+	if err != nil {
+		log.Errorf("error syncing config map: %s", err)
+		c.queue.AddRateLimited(key)
+		return true
+	}
+
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *CertificateController) syncNamespace(ns string) error {
+	conduitConfigMap, err := c.k8sAPI.CM().Lister().ConfigMaps(c.namespace).
+		Get(pkgK8s.CertificateBundleName)
+	if apierrors.IsNotFound(err) {
+		log.Warnf("configmap [%s] not found in namespace [%s]",
+			pkgK8s.CertificateBundleName, c.namespace)
+		return nil
```

Comment:
curious why we want to swallow this error?

related, i'm seeing these warning in the logs, expected?
```
time="2018-06-14T20:21:48Z" level=warning msg="configmap [conduit-ca-bundle] not found in namespace [conduit]"
time="2018-06-14T20:31:48Z" level=warning msg="configmap [conduit-ca-bundle] not found in namespace [conduit]"
time="2018-06-14T20:31:48Z" level=warning msg="configmap [conduit-ca-bundle] not found in namespace [conduit]"
time="2018-06-14T20:41:47Z" level=warning msg="configmap [conduit-ca-bundle] not found in namespace [conduit]"
time="2018-06-14T20:51:46Z" level=warning msg="configmap [conduit-ca-bundle] not found in namespace [conduit]"
```
probably related, i'm not seeing any configmaps getting created, am i missing someting?
```bash
$ kubectl --all-namespaces=true get cm
NAMESPACE     NAME                                 DATA      AGE
conduit       grafana-config                       3         52m
conduit       prometheus-config                    1         52m
kube-public   cluster-info                         2         2d
kube-system   extension-apiserver-authentication   6         2d
kube-system   kube-proxy                           2         2d
kube-system   kubeadm-config                       1         2d
```